packages/react-conformance/src/utils/merge.tsx

```diff
@@ -0,0 +1,41 @@
+import * as _ from 'lodash';
+
+// eslint-disable @typescript-eslint/no-explicit-any
+const isObject = (o: any) => o !== null && typeof o === 'object' && !Array.isArray(o);
+
+// eslint-disable @typescript-eslint/no-explicit-any
+const _merge = (obj1: any, obj2: any, mergeArrayPrimitive: boolean) => {
+  const keys: string[] = Object.getOwnPropertyNames(obj2);
+  keys.forEach(k => {
+    if (obj1 && !obj1.hasOwnProperty(k)) {
+      obj1[k] = obj2[k];
+    } else if (isObject(obj2[k])) {
+      _merge(obj1[k], obj2[k], mergeArrayPrimitive);
+    } else if (Array.isArray(obj2[k])) {
+      if (mergeArrayPrimitive) {
+        obj1[k] = (Array.isArray(obj1[k]) ? obj1[k] : [obj1[k]]).concat(obj2[k]);
+      } else if (Array.isArray(obj1[k])) {
+        obj1[k] = obj1[k].concat(obj2[k]);
+      }
+    }
+  });
+};
+
+/**
+ * Merge function with specific type
+ * @param mergeArrayPrimitive Merge array with primitive value, i.e.:
+ *  a { ob: ['foo', 'bar'] }
+ *  b { ob: 'a' }
+ *  merge(true, a, b) === { ob: ['foo', 'bar', 'a'] }
+ * @param objs
+ */
+export function merge<T = Object>(mergeArrayPrimitive: boolean, ...objs: Object[]): T {
```

Comment:
Why is merging arrays opt-in? At least to me it seems like that should be the default behavior, with no need to override it in this case.

Reply:
Arrays are always merged. This flag is so that when you have something like an integer or a value that isn't an array but the field with the same name in the second object is an array  you can decide if you want to merge the array with the single value. Something like:

`
obj1 = {
  foo: 1
}
obj2 = {
  foo: [2, 3, 4]
}

merge(true, obj1, obj2) === { foo: [1, 2, 3, 4] }
merge(false, obj1, obj2) === { foo: 1 }
`

Reply:
I thought this could be useful but I can remove it if it makes sense.

Reply:
Hmm interesting. That's a tricky case to know what the "right" thing to do is. But it's also not something we're very likely to encounter here, so it's probably okay to take a simple approach: if the first item and second item are different types, the second entirely overwrites the first.